sunshine-core/src/main/java/io/github/tatools/sunshine/core/VerboseRegex.java

```diff
@@ -0,0 +1,38 @@
+package io.github.tatools.sunshine.core;
+
+/**
+ * The implementation allows printing of encapsulated condition to the standard output. The printing will happen only
+ * once before the first usage of {@link #applicable(String)}.
+ *
+ * @author Dmytro Serdiuk (dmytro.serdiuk@gmail.com)
+ * @version $Id$
+ * @since 2018-06-09
+ */
+public final class VerboseRegex implements Condition {
+
+    private final RegexCondition regexCondition;
+    private final boolean[] say = new boolean[]{true};
```

Comment:
it looks like you don't need  an array at all.
As I understood, this variable is used to switch between 2 variants of regex - string or Pattern formats.
 It might be better following:
private final boolean isRegex = true;
....
if(isRegex){
System.out.println("...")
}
return...

Reply:
@IVKlymenko let's continue on this thread - https://github.com/tatools/sunshine/pull/140#discussion_r194527943